scripts/cf_subshell_scoped_login

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+set -eu
+
+if [ $# -lt 1 ]; then
+  echo "Usage $0 <target>" 1>&2
+  exit 1
+fi
+
+TARGET=${1}
+
+case $TARGET in
+  prod)
+    API_URL="https://api.cloud.service.gov.uk"
+    ;;
+  staging)
+    API_URL="https://api.staging.cloudpipeline.digital"
+    ;;
+  *)
+    echo "Unrecognised target '${TARGET}'" 1>&2
+    exit 1
+    ;;
+esac
+
+TMPDIR=${TMPDIR:-/tmp}
+CF_HOME=$(mktemp -d "${TMPDIR}/cf_home.XXXXXX")
+cleanup() {
+  echo "Cleaning up temporary CF_HOME..."
+  cf logout || true
+  rm -r "${CF_HOME}"
+}
+trap 'cleanup' EXIT
+
+export CF_HOME
+
+cf api "${API_URL}"
+cf login --sso
+
+echo
+echo "You are now in a subshell with CF_HOME set to ${CF_HOME}"
+echo "This will be cleaned up when this shell is closed."
+echo
+bash -il
```

Comment:
I normally use zsh instead of bash. Could this preserve the user's choice of shell?

Reply:
Good call. I've pushed another commit to use `$SHELL` instead.